test/lib/sdkdriver/test/unit/spec/transport/websocket/client.ts

```diff
@@ -0,0 +1,157 @@
+import WSClientTransport from '../../../../../src/transport/websocket/client';
+import { WebSocket } from '../../../mocks';
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+
+describe('WSClientTransport', () => {
+  describe('#close', () => {
+    let closeEventPromise: Promise<void>;
+    let transport: WSClientTransport;
+    let wsClient: WebSocket;
+
+    before(async () => {
+      transport = new WSClientTransport('foo', { WebSocket });
+      const openPromise: Promise<void> = transport.open();
+      wsClient = transport._wsClient;
```

Comment:
Would prefer if we accessed the `wsClient` from the WebSocket mock we inject instead of the private property on the WSClientTransport, since the latter could change.